test/system/surplus-for-auction.test.js

```diff
@@ -0,0 +1,186 @@
+      rewardsManager.address
+    )
+    await assetPool.deployed()
+    await underwriterToken.transferOwnership(assetPool.address)
+
+    const SignerBondsSwapStrategy = await ethers.getContractFactory(
+      "SignerBondsEscrow"
+    )
+    signerBondsSwapStrategy = await SignerBondsSwapStrategy.deploy()
+    await signerBondsSwapStrategy.deployed()
+
+    const Auction = await ethers.getContractFactory("Auction")
+
+    const masterAuction = await Auction.deploy()
+    await masterAuction.deployed()
+
+    const CoveragePool = await ethers.getContractFactory("CoveragePool")
+    coveragePool = await CoveragePool.deploy(assetPool.address)
+    await coveragePool.deployed()
+    await assetPool.transferOwnership(coveragePool.address)
+
+    const RiskManagerV1 = await ethers.getContractFactory("RiskManagerV1")
+    riskManagerV1 = await RiskManagerV1.deploy(
+      tbtcToken.address,
+      tbtcDepositTokenAddress,
+      coveragePool.address,
+      signerBondsSwapStrategy.address,
+      masterAuction.address,
+      auctionLength,
+      bondAuctionThreshold
+    )
+    await riskManagerV1.deployed()
+
+    await coveragePool
+      .connect(governance)
+      .approveFirstRiskManager(riskManagerV1.address)
+
+    tbtcDeposit = await ethers.getContractAt("IDeposit", depositAddress)
+    tbtcDeposit1 = await ethers.getContractAt("IDeposit", depositAddress1)
+
+    bidder = await impersonateAccount(bidderAddress)
+    bidder1 = await impersonateAccount(bidderAddress1)
+  })
+
+  describe("when buying auction with surplus TBTC funds", () => {
+    let surplusTx
+
+    before(async () => {
+      await riskManagerV1.notifyLiquidation(tbtcDeposit.address)
+
+      const auctionAddress = await riskManagerV1.depositToAuction(
+        tbtcDeposit.address
+      )
+
+      const auction = new ethers.Contract(auctionAddress, Auction.abi, bidder)
+      const bidderTake = lotSize.div(2) // 5 / 2 = 2.5 TBTC
+      await tbtcToken.connect(bidder).approve(auction.address, bidderTake)
+
+      // bidder takes a partial offer on a cov pool auction
+      await auction.takeOffer(bidderTake)
+    })
+
+    it("should have TBTC funds for partially selling an auction for deposit 0x849..", async () => {
+      const tbtcSurplus = await tbtcToken.balanceOf(riskManagerV1.address)
+      expect(tbtcSurplus).to.be.equal(lotSize.div(2))
+    })
+
+    it("should buy deposit 0x849.. outside the coverage pool", async () => {
+      await tbtcToken.connect(bidder1).approve(tbtcDeposit.address, lotSize)
+      // buying 0x849.. deposit outside the coverage pool
+      await tbtcDeposit.connect(bidder1).purchaseSignerBondsAtAuction()
+
+      await riskManagerV1.notifyLiquidated(tbtcDeposit.address)
+
+      expect(await tbtcDeposit.currentState()).to.equal(11) // LIQUIDATED
+    })
+
+    it("should buy deposit 0xfc9.. with surplus TBTC without opening an auction", async () => {
+      expect(await tbtcDeposit1.currentState()).to.equal(10) // IN LIQUIDATION
```

Comment:
Isn't it a good candidate for the `test initial state` section we have in other tests?

Reply:
Done